.github/workflows/boulder-ci.yml

```diff
@@ -0,0 +1,100 @@
+# Boulder CI test suite workflow
+
+name: Boulder CI
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  #  Boulder_ci_tests job. This looks like a single job, but the matrix
+  #  items will mulitply it. For example every entry in the
+  #  BOULDER_TOOLS_TAG list will run with every test. If there were two
+  #  tags and 5 tests there would be 10 jobs run.
+  boulder_ci_tests:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-20.04
+
+    strategy:
+      # When set to true, GitHub cancels all in-progress jobs if any matrix job fails. Default: true
+      fail-fast: true
```

Comment:
It can be useful to see results from other jobs. For example, if the unit tests fail quickly, that may cause the integration tests to be cancelled, so then you fix the unit test failure and upload a new commit... only to find that the integration tests are failing and _would_ have failed before if they hadn't been cancelled.

Obviously there are advantages to cancellation too (namely freeing up resources for other PRs) but by default I'd lean towards no cancelling on first failure.

Reply:
True. Since we aren't waiting for 40 minutes on these tests, I think it doesn't hurt to flip it to false.